diff --git a/lib/librarian/source/git/repository.rb b/lib/librarian/source/git/repository.rb
index cba083a..961e2ac 100644
--- a/lib/librarian/source/git/repository.rb
+++ b/lib/librarian/source/git/repository.rb
@@ -21,7 +21,7 @@ def bin
           end
 
           def git_version
-            command = %W[#{bin} version --silent]
+            command = %W[#{bin} version]
             Posix.run!(command).strip =~ /\Agit version (\d+(\.\d+)*)/ && $1
           end
         end
