--- all/puppetdb-2.0.0/tasks/tar.rake.orig	2014-08-27 19:44:25.883778453 +0000
+++ all/puppetdb-2.0.0/tasks/tar.rake	2014-08-27 19:45:13.405689867 +0000
@@ -25,8 +25,8 @@
   File.open(File.join(workdir,'version'), File::CREAT|File::TRUNC|File::RDWR, 0644) do |f|
     f.puts @version
   end
-  mv "#{workdir}/ext/files/debian", "#{workdir}/ext"
-  mv "#{workdir}/ext/files/#{@name}.spec", "#{workdir}/ext/redhat/"
+  #mv "#{workdir}/ext/files/debian", "#{workdir}/ext"
+  #mv "#{workdir}/ext/files/#{@name}.spec", "#{workdir}/ext/redhat/"
   cp_pr "puppet", "#{workdir}/ext/master"
   mkdir_p "pkg"
   pkg_dir = File.expand_path(File.join(".", "pkg"))
--- all/puppetdb-2.0.0/tasks/install.rake.orig	2014-08-27 19:45:29.369323974 +0000
+++ all/puppetdb-2.0.0/tasks/install.rake	2014-08-27 19:48:23.461316203 +0000
@@ -108,6 +108,14 @@
     cp_p "ext/files/puppetdb.env", "#{DESTDIR}/#{@libexec_dir}/#{@name}.env"
     cp_p "ext/files/systemd/#{@name}.service", "#{DESTDIR}/usr/lib/systemd/system"
     chmod 0644, "#{DESTDIR}/usr/lib/systemd/system/#{@name}.service"
+  elsif @osfamily == "gentoo"
+    mkdir_p "#{DESTDIR}/etc/init.d"
+    mkdir_p "#{DESTDIR}/etc.conf.d"
+    mkdir_p "#{DESTDIR}/usr/lib/systemd/system"
+    cp_p "ext/files/#{@name}.debian.init", "#{DESTDIR}/etc/init.d/#{@name}"
+    cp_p "ext/files/puppetdb.default.systemd", "#{DESTDIR}/etc/conf.d/#{@name}"
+    cp_p "ext/files/systemd/#{@name}.service", "#{DESTDIR}/usr/lib/systemd/system/"
+    chmod 0755, "#{DESTDIR}/etc/init.d/#{@name}"
   else
     raise "Unknown or unsupported osfamily: #{@osfamily}"
   end
--- all/puppetdb-2.0.0/Rakefile.orig	2014-08-27 19:43:02.638685787 +0000
+++ all/puppetdb-2.0.0/Rakefile	2014-08-27 19:43:28.426094630 +0000
@@ -42,7 +42,7 @@
     # If we're building PE, we need to set the project name to pe-puppetdb
     Pkg::Config.project = "pe-puppetdb"
   end
-  @version = Pkg::Config.version
+  @version = "2.0.0"
 else
   begin
     %x{which git >/dev/null 2>&1}
--- all/puppetdb-2.0.0/ext/templates/puppetdb.service.erb.orig	2014-08-27 19:43:52.460543921 +0000
+++ all/puppetdb-2.0.0/ext/templates/puppetdb.service.erb	2014-08-27 19:44:12.254090836 +0000
@@ -4,7 +4,7 @@
 
 [Service]
 Type=simple
-EnvironmentFile=/etc/sysconfig/<%= @name %>
+EnvironmentFile=/etc/conf.d/<%= @name %>
 User=<%= @name %>
 PIDFile=/var/run/<%= @name %>/<%= @name %>.pid
 <%- if @pe -%>
