--- all/puppetdb-1.6.3/tasks/tar.rake.bak	2014-04-19 15:49:08.706725692 +0000
+++ all/puppetdb-1.6.3/tasks/tar.rake	2014-04-19 15:49:12.323291821 +0000
@@ -25,8 +25,8 @@
   File.open(File.join(workdir,'version'), File::CREAT|File::TRUNC|File::RDWR, 0644) do |f|
     f.puts @version
   end
-  mv "#{workdir}/ext/files/debian", "#{workdir}/ext"
-  mv "#{workdir}/ext/files/#{@name}.spec", "#{workdir}/ext/redhat/"
+  #mv "#{workdir}/ext/files/debian", "#{workdir}/ext"
+  #mv "#{workdir}/ext/files/#{@name}.spec", "#{workdir}/ext/redhat/"
   cp_pr "puppet", "#{workdir}/ext/master"
   mkdir_p "pkg"
   pkg_dir = File.expand_path(File.join(".", "pkg"))
--- all/puppetdb-1.6.3/tasks/install.rake.orig	2014-04-24 20:48:25.933989759 +0000
+++ all/puppetdb-1.6.3/tasks/install.rake	2014-04-24 20:50:37.966945605 +0000
@@ -100,6 +100,14 @@
     cp_p "ext/files/puppetdb.openbsd.init", "#{DESTDIR}/etc/rc.d/#{@name}.rc"
     cp_p "ext/files/puppetdb.env", "#{DESTDIR}/#{@libexec_dir}/#{@name}.env"
     chmod 0755, "#{DESTDIR}/etc/rc.d/#{@name}.rc"
+  elsif @osfamily == "gentoo"
+    mkdir_p "#{DESTDIR}/etc/init.d"
+    mkdir_p "#{DESTDIR}/etc/conf.d"
+    mkdir_p "#{DESTDIR}/usr/lib/systemd/system"
+    cp_p "ext/files/#{@name}.debian.init", "#{DESTDIR}/etc/init.d/#{@name}"
+    cp_p "ext/files/puppetdb.default.systemd", "#{DESTDIR}/etc/conf.d/#{@name}"
+    cp_p "ext/files/systemd/#{@name}.service", "#{DESTDIR}/usr/lib/systemd/system"
+    chmod 0755, "#{DESTDIR}/etc/init.d/#{@name}"
   else
     raise "Unknown or unsupported osfamily: #{@osfamily}"
   end
--- all/puppetdb-1.6.3/Rakefile.bak	2014-04-19 17:45:32.611575220 +0000
+++ all/puppetdb-1.6.3/Rakefile	2014-04-19 17:45:46.587854388 +0000
@@ -42,7 +42,7 @@
     # If we're building PE, we need to set the project name to pe-puppetdb
     Pkg::Config.project = "pe-puppetdb"
   end
-  @version = Pkg::Config.version
+  @version = "1.6.3"
 else
   begin
     %x{which git >/dev/null 2>&1}
--- all/puppetdb-1.6.3/ext/templates/puppetdb.service.erb.orig	2014-04-24 20:41:41.505338690 +0000
+++ all/puppetdb-1.6.3/ext/templates/puppetdb.service.erb	2014-04-24 20:42:10.351196448 +0000
@@ -4,7 +4,7 @@
 
 [Service]
 Type=simple
-EnvironmentFile=/etc/sysconfig/<%= @name %>
+EnvironmentFile=/etc/conf.d/<%= @name %>
 User=<%= @name %>
 PIDFile=/var/run/<%= @name %>/<%= @name %>.pid
 <%- if @pe -%>
