--- setup.py	2013-03-09 18:51:25.895110688 +0000
+++ setup.py	2013-03-09 18:51:33.271778090 +0000
@@ -42,7 +42,6 @@
         'jinja2',
         'yolk',
         'pygments',
-        'argparse',
         'jaxml>=3.02',
         'metagen',  # metadata.xml
         'sphinxcontrib-googleanalytics',
--- g_pypi.egg-info/requires.txt	2013-03-09 18:03:56.631755286 +0000
+++ g_pypi.egg-info/requires.txt	2013-03-09 18:04:03.341749821 +0000
@@ -2,7 +2,6 @@
 jinja2
 yolk
 pygments
-argparse
 jaxml>=3.02
 metagen
 sphinxcontrib-googleanalytics

